packages/@azure/servicebus/data-plane/test/renewLockSessions.spec.ts

```diff
@@ -431,20 +434,37 @@ async function testAutoLockRenewalConfigBehavior(
         // Sleeping...
         await delay(options.delayBeforeAttemptingToCompleteMessageInSeconds * 1000);
 
+        // Following code gets executed in background but since session link is closed,
+        // errors from here if any don't get surfaced.
+        // So the code below gets executed without errors but is not useful for tests
```

Comment:
Are we saying that the `catch` block at https://github.com/Azure/azure-sdk-for-js/blob/master/packages/%40azure/servicebus/data-plane/lib/session/messageSession.ts#L563 is never reached?

Reply:
As discussed, the error thrown from the users onMessage handler is getting eaten up in the catch block 
https://github.com/Azure/azure-sdk-for-js/blob/master/packages/%40azure/servicebus/data-plane/lib/session/messageSession.ts#L563 because the `isOpen()` will return `false` as the session receiver is closed.

In this test, we want to ensure 2 things
- User's onError handler gets called when SessionLockLostError surfaces (which is being tested just fine here)
- `.complete()` returns `SessionLockLostError` (this, we are failing to test here because of the error getting eaten up).

For the second point above, lets just move the `.complete()` out of the call back, and into the line just before we close the session receiver.